synapse/federation/federation_server.py

```diff
@@ -710,7 +710,7 @@ async def on_send_join_request(
         state_event_ids: Collection[str]
         servers_in_room: Optional[Collection[str]]
         if caller_supports_partial_state:
-            summary = await self.store.get_room_summary(room_id)
+            summary = await self.store.get_room_summary(room_id)  # type: ignore[synapse-@cached-mutable]
```

Comment:
I'm confused why we need to do a type ignore at call sites?

Reply:
Because the context the plug-in gets is the call sites, not the definition site. We've been unable to figure out how to get the errors to appear on the definition locations.

Reply:
I got nerd sniped https://github.com/matrix-org/synapse/commit/745ad617188a69fd19336a93053ac8332d9b5581